bot/worker.py

```diff
@@ -65,116 +74,290 @@ def process_replyable(reddit, replyable):
+    :param text: The processed body/title text
+    :return: None
+    """
+    hero_name, text = text.split('::', 1)
+    hero_id = db_api.get_hero_id_by_name(hero_name=hero_name)
+    link, _ = db_api.get_link_for_response(processed_text=text, hero_id=hero_id)
+    create_and_add_reply(replyable=replyable, response_url=link, hero_id=hero_id)
+
+
+def is_flair_specific_response(replyable, text):
+    """Method that checks if response for hero in author's flair and text exists.
+
+    :param replyable: The comment/submission on reddit
+    :param text: The processed body/title text
+    :return: True if the response for author's flair's hero was found, else False
     """
     hero_id = db_api.get_hero_id_by_flair_css(flair_css=replyable.author_flair_css_class)
     if hero_id:
-        link, hero_id = db_api.get_link_for_response(processed_text=processed_text, hero_id=hero_id)
+        link, _ = db_api.get_link_for_response(processed_text=text, hero_id=hero_id)
         if link:
-            reply = create_reply(replyable=replyable, response_url=link, hero_id=hero_id)
-            replyable.reply(reply)
-            logger.info("Added: " + replyable.fullname)
             return True
     return False
 
 
-def add_regular_reply(replyable, processed_text):
-    """Method to create response for given replyable.
-    In case of multiple matches, it used to sort responses in descending order of heroes, but now it's random.
+def add_flair_specific_reply(replyable, text):
+    """Method to add a author's flair specific reply to the comment/submission.
+
+    :param replyable: The comment/submission on reddit
+    :param text: The processed body/title text
+    :return: None
+    """
+    hero_id = db_api.get_hero_id_by_flair_css(flair_css=replyable.author_flair_css_class)
+    link, _ = db_api.get_link_for_response(processed_text=text, hero_id=hero_id)
+    create_and_add_reply(replyable=replyable, response_url=link, hero_id=hero_id)
+
+
+def is_update_request(reddit, replyable, text):
+    """Method to check whether the comment in a request to update existing response.
+    Only works if
+    * Comment beings with "try"
+    * Comment ends with valid hero name
+    * Comment was made as a reply to bot's comment
+    * Given hero has the original response
+    * Comment was added by OP, who made the original request for the response.
+    * Root/Original comment/submission was not hero specific response.
+
+    The comment tree would look something like below, where root(original) replyable can be Comment or Submission.
+    Only valid case is c3.
+    c1/s1 user: Foo
+        c2 bot: "Foo" response by Bar hero
+           c3 user: Try Bar2
+           c4 other_user: Try Bar2
 
+    :param reddit: The reddit account instance
     :param replyable: The comment/submission on reddit
-    :param processed_text: The processed body text
+    :param text: The processed body/title text
+    :return: True if this is a valid request, else False
```

Comment:
if -- else sounded more natural to me, maybe because it's close how to write program, but "otherwise" does seem more fitting in this situation. maybe I can start using "otherwise" from now on, or do you want the past occurences changed as well?

Reply:
Changes only in the future seem reasonable, it is not that crucial :wink: